storage/mroonga/vendor/groonga/lib/hash.c

```diff
@@ -1865,12 +1865,12 @@ grn_tiny_hash_calculate_entry_size(uint32_t key_size, uint32_t value_size,
 {
   uint32_t entry_size;
   if (flags & GRN_OBJ_KEY_VAR_SIZE) {
-    entry_size = (uintptr_t)((grn_tiny_hash_entry *)0)->value + value_size;
+    entry_size = (uintptr_t)offsetof(grn_tiny_hash_entry, value) + value_size;
   } else {
     if (key_size == sizeof(uint32_t)) {
-      entry_size = (uintptr_t)((grn_plain_hash_entry *)0)->value + value_size;
+      entry_size = (uintptr_t)offsetof(grn_plain_hash_entry, value) + value_size;
     } else {
-      entry_size = (uintptr_t)((grn_rich_hash_entry *)0)->key_and_value
+      entry_size = (uintptr_t)offsetof(grn_rich_hash_entry, key_and_value)
```

Comment:
you likely don'y need `uintptr_t` casts here either

Reply:
Except for the db.c case which results in the following without it:
```
/source/storage/mroonga/vendor/groonga/lib/db.c:972:17: error: incompatible pointer to integer conversion assigning to 'uint32_t' (aka 'unsigned int') from 'double *' [-Wint-conversion]
  972 |     *value_size =GRN_RSET_SUBRECS_NTH(offsetof(grn_rset_recinfo, subrecs),
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  973 |                                                 *subrec_size, max_n_subrecs);
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Removed other cases.